howso/direct/client.py

```diff
@@ -4896,6 +4971,155 @@ def get_trainee_session_training_indices(
+        method : str, name of method to execute
+        payload : dict, parameters specific to the method being called
+        child_name_path : list of str, optional
+            List of strings specifying the user-friendly path of the child
+            subtrainee for execution of method.
+        child_id : str, optional
+            Unique id of child trainee to execute method. Ignored if
+            child_name_path is specified.
+        as_external : bool
+            Applicable only to 'load' and 'save' methods and if specifying
+            child_name_path or child_id.
+            For 'save', stores the child out as an independent trainee and removes
+            it as a contained entity.
+            For 'load' updates hierarchy by adding the child as an independently
+            stored trainee to the hierarchy without loading the trainee as a
+            subtrainee.
+        load_external_trainee_id : str, optional
+            Trainee id of trainee being loaded, must be specified only
+            when method is 'load' and as_external is true.
+        trainee_id : str
+            The id of the Trainee to execute methods on.
+
+        Returns
+        -------
+        object
+            Whatever output the executed method returns.
+        """
+        self._auto_resolve_trainee(trainee_id)
+
+        # when creating child trainees, ensure they have a uid specified
+        if method == 'create_trainee':
+            if 'trainee_id' not in payload:
+                payload['trainee_id'] = str(uuid.uuid4())
+
+        result = self.howso.execute(
+            trainee_id,
+            method=method,
+            as_external=as_external,
+            child_id=child_id,
+            child_name_path=child_name_path,
+            payload=payload,
+            load_external_trainee_id=load_external_trainee_id
+        )
+
+        if method in ('get_cases', 'retrieve_extreme_cases_for_feature'):
+            if result is None:
+                result = dict()
+            return Cases(features=result.get('features'),
+                                    cases=result.get('cases'))
+
+        elif method in ['react', 'batch_react']:
+            # Action values and features should always be defined
+            if not result.get('action_values'):
+                result['action_values'] = []
+            if not result.get('action_features'):
+                result['action_features'] = []
+            if method=='react':
+                result['action_values'] = [result['action_values']]
+            result = internals.format_react_response(result) #, method=='react')
+            result = Reaction(result.get('action'), result.get('details'))
+
+        elif method in ['react_series', 'batch_react_series']:
+            ret = dict()
+            result = replace_doublemax_with_infinity(result)
+            ret['action_features'] = result.pop('action_features') or []
+            ret['action'] = result.pop('action_values')
```

Comment:
This will raise if the key 'action_values' doesn't exist in `result`. Is this intentionally left to be raised?

If not, consider something like: `ret['action_values'] = result.pop('action_values, [])`

Reply:
it'll always be there, this code is copied from elsewhere